### PR TITLE
final - Refactor HTTP responses

### PR DESCRIPTION
**Pull request by vdedios**
solves #97 #98 #99

### CODE REVIEW

File: srcs/app/http/response/HttResponseFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include <HttpResponseFactory.hpp>
```
Comment (Damrod) [inline srcs/app/http/response/HttResponseFactory.cpp:1]: This filename is missing the 'p' after the Htt.

HttResponseFactory.cpp -> HttpResponseFactory.cpp

File: srcs/app/http/response/HttErrorResponse.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include "HttpErrorResponse.hpp"
```
Comment (Damrod) [inline srcs/app/http/response/HttErrorResponse.cpp:1]: This filename is missing the 'p' after the Htt.

HttResponseFactory.cpp -> HttpResponseFactory.cpp

**Review by Damrod — APPROVED** [review]
Yo diría que no pierdas ni un minuto más con CGI, porque la ultima version [que tenemos](https://github.com/42AGV/webserv/pull/102/files) (que gbudau hizo), no se parece en nada a lo que tenemos en master...
Por otra parte, como habrás obsevado, tienes varios tests, unitarios y de integración, que fallan, pero a lo mejor lo pretendes arreglar en la rama base?

File: srcs/app/http/response/HttResponseFactory.cpp (hunk 1 above)
Comment (vdedios) [reply srcs/app/http/response/HttResponseFactory.cpp:1]: Good catch! I renamed several files and this could be due to it. Thanks!

**Comment by vdedios** [comment]
Yes, I am aware of broken tests. Will fix them now until @gbudau reviews this PR. The should be merged with everything working

**Review by gbudau — APPROVED** [review]
Impressive work, thank you!

Valgrind reports a few errors.
<details>
  <summary>Click to expand</summary>
```
==12599== Memcheck, a memory error detector
==12599== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==12599== Using Valgrind-3.16.1 and LibVEX; rerun with -h for copyright info
==12599== Command: ./webserv
==12599== 
==12599== Conditional jump or move depends on uninitialised value(s)
==12599==    at 0x12B7F5: FDsets::addToReadSet(int) (FDsets.cpp:18)
==12599==    by 0x1272CE: WebServer::PopulateServers_() (WebServer.cpp:51)
==12599==    by 0x126D7F: WebServer::WebServer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (WebServer.cpp:5)
==12599==    by 0x10D83C: main (main.cpp:15)
==12599== 
^C==12599== 
==12599== Process terminating with default action of signal 2 (SIGINT)
==12599==    at 0x54F6DD7: select (select.c:41)
==12599==    by 0x126F50: WebServer::Run() (WebServer.cpp:21)
==12599==    by 0x10D84B: main (main.cpp:16)
==12599== 
==12599== HEAP SUMMARY:
==12599==     in use at exit: 123,225,182 bytes in 497 blocks
==12599==   total heap usage: 5,811 allocs, 5,314 frees, 926,893,690 bytes allocated
==12599== 
==12599== 1,999 (192 direct, 1,807 indirect) bytes in 3 blocks are definitely lost in loss record 276 of 285
==12599==    at 0x4C3252A: operator new(unsigned long) (vg_replace_malloc.c:342)
==12599==    by 0x13AD5E: HttpResponseFactory::createHttpPostResponse_() (HttpResponseFactory.cpp:57)
==12599==    by 0x13AB82: HttpResponseFactory::Response() (HttpResponseFactory.cpp:40)
==12599==    by 0x12B4AC: Connection::SendResponse() (Connection.cpp:41)
==12599==    by 0x1290F3: Server::SendResponse(int) (Server.cpp:85)
==12599==    by 0x127621: WebServer::HandleWriteSocket_(int) (WebServer.cpp:92)
==12599==    by 0x1270E0: WebServer::Run() (WebServer.cpp:33)
==12599==    by 0x10D84B: main (main.cpp:16)
==12599== 
==12599== 31,307 (2,048 direct, 29,259 indirect) bytes in 32 blocks are definitely lost in loss record 284 of 285
==12599==    at 0x4C3252A: operator new(unsigned long) (vg_replace_malloc.c:342)
==12599==    by 0x13AD02: HttpResponseFactory::createHttpGetResponse_() (HttpResponseFactory.cpp:53)
==12599==    by 0x13AB82: HttpResponseFactory::Response() (HttpResponseFactory.cpp:40)
==12599==    by 0x12B4AC: Connection::SendResponse() (Connection.cpp:41)
==12599==    by 0x1290F3: Server::SendResponse(int) (Server.cpp:85)
==12599==    by 0x127621: WebServer::HandleWriteSocket_(int) (WebServer.cpp:92)
==12599==    by 0x1270E0: WebServer::Run() (WebServer.cpp:33)
==12599==    by 0x10D84B: main (main.cpp:16)
==12599== 
==12599== LEAK SUMMARY:
==12599==    definitely lost: 2,240 bytes in 35 blocks
==12599==    indirectly lost: 31,066 bytes in 90 blocks
==12599==      possibly lost: 0 bytes in 0 blocks
==12599==    still reachable: 123,191,876 bytes in 372 blocks
==12599==         suppressed: 0 bytes in 0 blocks
==12599== Reachable blocks (those to which a pointer was found) are not shown.
==12599== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==12599== 
==12599== Use --track-origins=yes to see where uninitialised values come from
==12599== For lists of detected and suppressed errors, rerun with: -s
==12599== ERROR SUMMARY: 3 errors from 3 contexts (suppressed: 0 from 0)
```
</details>

**Comment by vdedios** [comment]
yes, I am aware also there are a few memory leaks, but I have identified them, will fix it in a commit. Thanks!!

**Pull request merged**